src/remotelog.lua

```diff
@@ -0,0 +1,221 @@
+-- that uses the log and a version number.
+-- </p>
+--
+function M.set_client_name(log_client_name)
+    M.log_client_name = log_client_name
+end
+
+local function start_high_resolution_timer()
+    if M.use_high_resolution_time then
+        M.start_nanos = socket.gettime()
+    end
+end
+
+local function get_level_name(level)
+    for k, v in pairs(levels) do
+        if v == level then
+            return k
+        end
+    end
+    error("E-LOG-1: Unable to determine log level name for level number " .. level .. ".")
+end
+
+---
+-- Open a connection to a remote log receiver.
+-- <p>
+-- This method allows connecting the log to an external process listening on a TCP port. The process can be on a remote
+-- host. If the connection cannot be established, the logger falls back to console logging.
+-- </p>
+--
+-- @param host remote host on which the logging process runs
+--
+-- @param port TCP port on which the logging process listens
+--
+function M.connect(host, port)
+    local tcp_socket = socket.tcp()
+    tcp_socket:settimeout(M.connection_timeout)
+    local ok, err = tcp_socket:connect(host, port)
+    local log_client_prefix = M.log_client_name and (M.log_client_name .. ": ") or ""
+    if ok then
+        M.socket_client = tcp_socket
+        M.info("%sConnected to log receiver listening on %s:%d with log level %s. Time zone is UTC%s.", log_client_prefix, host, port,
+            get_level_name(M.level), os.date("%z"))
+    else
+        print(log_client_prefix .. "W-LOG-2: Unable to open socket connection to " .. host .. ":" .. port
+            .. " for sending log messages. Falling back to console logging with log level " .. get_level_name(M.level)
+            .. ". Timezone is UTC" .. os.date("%z") .. ". Caused by: " .. err)
+    end
+end
+
+---
+-- Close the connection to the remote log receiver.
+--
+function M.disconnect()
+    if(M.socket_client) then
+        M.socket_client:close()
+    end
+end
+
+---
+-- Set the log level.
+--
+-- @param level_name name of the log level, one of: FATAL, ERROR, WARN, INFO, CONFIG, DEBUG, TRACE
+--
+function M.set_level(level_name)
+    local level = levels[level_name]
+    if not level then
+        M.warn('W-LOG-1: Attempt to set illegal log level "%s". Pick one of: NONE, FATAL, ERROR, WARN, INFO, CONFIG,'
+        .. ' DEBUG, TRACE. Falling back to level INFO.', level_name)
+        M.level = levels.INFO
+    else
+        M.level = level
+    end
+end
+
+---
+-- Write to a socket, print or discard message.
+-- <p>
+-- If a socket connection is established, this method writes to that socket. Otherwise if the global print function
+-- exists (e.g. in a unit test) falls back to logging via <code>print()</code>.
+-- </p><p>
+-- Exasol removed print() in it's Lua implementation, so there is no fallback on a real Exasol instance. You either use
+-- remote logging or messages are discarded immediately.
```

Comment:
I could not verify that.

I just tried it again with the latest version and get the following error:
```
SQL Error [22001]: Lua Error "attempt to call global 'print' (a nil value)" caught in script "JB"."JB_TEST" at line 107
```